### PR TITLE
test(maintenance): switch tracer to vitest

### LINKED ISSUE #3284: Maintenance: migrate test runner to vitest for tracer package
### Summary

To continue the migration from Jest to Vitest we need to move over the tests for the `tracer` package to use the new test runner.

### Why is this needed?

So that we can continue the migration as well as improve the overall performance and consistency of the tests.

### Which area does this relate to?

Parser

### Solution

_No response_

### Acknowledgment

- [x] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR moves the unit and end to end tests for the Tracer package to use vitest as test runner.

This action is part of the ongoing migration from Jest to Vitest, and the vast majority of the changes in the tests focus on changing the imports and using Vitest-specific APIs.

The file with most changes is the `packages/tracer/tests/unit/Provider.test.ts`, which has a module import mock for the `aws-xray-sdk-node` package. Because Jest and Vitest mock imports slightly differently, I had to make more changes in this file than in others.

I also have removed some redundant test cases while maintaining the same level of coverage.

Two extra files from other packages were touched, respectively for:
- `setupEnv.ts` under `packages/testing` was modified to include two environment variables needed for Tracer during tests.
- One Logger test case was modified to take in account the new value of an env variable that was modified as part of the point above.

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #3284

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Integration tests passing on this branch: https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/11669742574/job/32492395223

File: packages/tracer/src/Tracer.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ class Tracer extends Utility implements TracerInterface {
    * @deprecated Use {@link captureAWSv3Client} instead.
    * @param service - AWS SDK v2 client
    */
-  public captureAWSClient<T>(service: T): T {
+  /* v8 ignore start */ public captureAWSClient<T>(service: T): T {
```
Comment (am29d) [inline packages/tracer/src/Tracer.ts:324]: What is the reason for v8 ignore directive? Deprecation?

File: packages/tracer/tests/unit/ProviderService.test.ts [hunk 2]
```diff
@@ -1,102 +1,94 @@
...
+vi.mock('aws-xray-sdk-core', async (importOriginal) => ({
+  default: {
+    ...(
+      await importOriginal<{
+        default: typeof import('aws-xray-sdk-core');
+      }>()
+    ).default,
+    ...mocks,
+  },
+}));
```
Comment (am29d) [inline packages/tracer/tests/unit/ProviderService.test.ts:38]: wow, this is an interesting level of sorcery

File: packages/tracer/tests/unit/Tracer.test.ts [hunk 3]
```diff
@@ -11,36 +6,29 @@ import {
...
-  [
-    name: string,
-    fcn: (subsegment?: Subsegment) => unknown,
-    parent?: Segment | Subsegment,
-  ]
->;
+/* vi.hoisted(() => {
+  process.env.POWERTOOLS_SERVICE_NAME = 'hello-world';
+  process.env.AWS_XRAY_LOGGING_LEVEL = 'silent';
+}); */
```
Comment (am29d) [inline packages/tracer/tests/unit/Tracer.test.ts:18]: Forgot to remove?

**Review by am29d — CHANGES_REQUESTED** [review]
Thanks for the PR! I have only few questions for clarification and we are good to merge.

File: packages/tracer/src/Tracer.ts (hunk 1 above)
Comment (dreamorosi) [reply packages/tracer/src/Tracer.ts:324]: Yes, correct. The method is marked as deprecated (L321) and so I just removed the test that was using it.

File: packages/tracer/tests/unit/ProviderService.test.ts (hunk 2 above)
Comment (dreamorosi) [reply packages/tracer/tests/unit/ProviderService.test.ts:38]: Yes, it's not very intuitive.

Here are [the docs for the Vitest mocking feature](https://vitest.dev/api/vi.html#mock-modules). Basically we are using Vitest to hijack the module resolution for a specific module and returning our own object.

In doing so however, we are not replacing the entire module but only some of it. To do so, we are importing the original module and destructuring its content in the object we return. Next we overwrite some of the properties that we want to provide a mock for (see `mocks` object).

This specific module has also an added complexity which is the fact that it only provides a `default` export, which is why we need to nest the mock object within a `default` property.

**Comment by dreamorosi** [comment]
@am29d thanks for the review, addressed your comments & ready for another round.

**Pull request merged**